src/test/java/org/openrewrite/java/migrate/javax/RemoveTemporalAnnotationTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.javax;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.DocumentExample;
+import org.openrewrite.InMemoryExecutionContext;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+class RemoveTemporalAnnotationTest implements RewriteTest {
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.parser(JavaParser.fromJavaVersion().classpathFromResources(new InMemoryExecutionContext(), "javax.persistence-api-2.2"))
+          .recipe(new RemoveTemporalAnnotation());
+    }
+
+    @Test
+    @DocumentExample
+    void removeTemporalAnnotation() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import javax.persistence.Temporal;
+              import javax.persistence.TemporalType;
+              import java.sql.Date;
+              import java.sql.Time;
+              import java.sql.Timestamp;
+
+              public class TemporalDates {
```

Comment:
Can you port over the following fields:

- TemporalType.TIMESTAMP used with `java.sql.Date`
- TemporalType.DATE used with `java.sql.Timestamp`
- TemporalType.TIME used with `java.sql.Timestamp`

Reply:
Actually I think the help we have for this issue is incorrect. It sounds like this recipes only needs to remove annotation for the following combinations:

- TemporalType.TIMESTAMP and java.sql.Timestamp
- TemporalType.DATE and java.sql.Date
- TemporalType.TIME and java.sql.Time

![Screenshot 2024-03-07 at 3 27 24 PM](https://github.com/openrewrite/rewrite-migrate-java/assets/48495887/0a0e1a69-754f-44ae-b3e6-aa16992b3202)

Reply:
I did notice the rule is overly broad, but the other cases that we remove the annotation would have been an invalid combination in either OpenJPA or EclipseLink anyway. So it would just fix the code error.

Reply:
For clarity, the current implementation handles/fixes all the cases except the ones that require a converter. 
Do we want to include the converter in this recipe?

Reply:
Let's not worry about the converter for now. But I'll create an issue to look into it later.